src/odbc/src/app/application_data_buffer.cpp

```diff
@@ -290,9 +291,9 @@ ConversionResult::Type ApplicationDataBuffer::PutStrToStrBuffer(
   bool isTruncated = false;
   if (inCharSize == 1) {
     if (outCharSize == 2 || outCharSize == 4) {
-      bytesWritten = utility::CopyUtf8StringToSqlWcharString(
-          reinterpret_cast< const char* >(value.c_str() + currentCellOffset),
-          reinterpret_cast< SQLWCHAR* >(dataPtr), buflen, isTruncated);
+      bytesWritten = utility::CopyWcharStringToSqlWcharString(
```

Comment:
I don't believe this function call is required. The conversion from UTF-8 -> SQLWCHAR already happened on line 266. We want to avoid any additional copying here.

Reply:
There is no bug in the code, only that it is doing extra copies than necessary. All the test cases should be covered, so we shouldn't need to create any new functions. Just rework the function to avoid the extra copying.

- *Windows*: The SQLWCHAR is defined as wchar_t (2-byte)
- *MacOS (iODBC)*: The SQLWCHAR is defined as wchar_t (4-byte)
- *Linux (unixODBC)*: The SQLWCHAR is defined as unsigned short (2-byte)